backend/layers/common/entities.py

```diff
@@ -0,0 +1,65 @@
+from dataclasses import dataclass
+from datetime import datetime
+from typing import List
+
+
+@dataclass
+class DatasetStatus:
+    status: str  # TODO: use an enum
+
+
+@dataclass
+class DatasetArtifact:
+    type: str
+    uri: str
+
+
+@dataclass
+class DatasetMetadata:
+    organism: str
+    tissue: str
+    assay: str
+    disease: str
+    sex: str
+    self_reported_ethnicity: str
+    development_stage: str
+    cell_type: str
+    cell_count: int
+
+
+@dataclass
+class Dataset:
+    id: str
+    status: DatasetStatus
+    metadata: DatasetMetadata
+    artifacts: List[DatasetArtifact]
```

Comment:
These last 3 should be in the `DatasetVersion`, right?

Reply:
See my comment about `Collection` (also applies here 😄 )